services/QuillLMS/app/models/concerns/public_progress_reports.rb

```diff
@@ -14,11 +14,11 @@ def last_completed_diagnostic
                   .first
     end
 
-    def activity_session_report(classroom_unit_id, user_id, activity_id)
-      classroom_unit = ClassroomUnit.find(classroom_unit_id)
-      unit_id = classroom_unit.unit_id
-      classroom_id = classroom_unit.classroom_id
-      if unit_id && activity_id && classroom_id
+    def activity_session_report(unit_id, classroom_id, user_id, activity_id)
+      if Activity.diagnostic_activity_ids.include?(activity_id.to_i)
```

Comment:
Not worth doing in this PR, I don't think, but the number of places we have this particular `include?` check makes me think that maybe we should have a an `Activity.is_diagnostic_id?()` function that does the `to_i` and `include?` check just to simplify all the call sites.  Something to consider for later.